### ISSUE TITLE
Improve the dashboard otlp experience

### ISSUE THREAD

**Issue by davidfowl** [prom3theu5/aspirational-manifests#177]
First a few concrete suggestions

We should set:

```
OTEL_SERVICE_NAME={resourcename}
OTEL_RESOURCE_ATTRIBUTES=service.instance.id={resourcename}
```

This is what it looks like right now:

<img width="1200" alt="image" src="https://github.com/prom3theu5/aspirational-manifests/assets/95136/48c66806-6b07-4e09-8b90-15dbd1ad3b0e">

We should also consider a default otel collector experience that routes telemetry to the dashboard. See https://github.com/practical-otel/opentelemetry-aspire-collector for more information on how this would work.

cc @JamesNK @martinjt

**Comment by martinjt** [comment]
`service.instance.id` shouldn't be the service name. It's supposed to be the subset of the service that this running instance is, so the guid is right.

The issue is that the dashboard is displaying it as the primary rather than the service name being the primary.

The instance id isn't actually relevant for the trace waterfall normally, only when you're doing more advanced debugging like searching or adding to a table.

Also keep in mind that `service.instance.id` is still marked as experimental at this stage. It's unlikely to change, but something to keep in mind.

**Comment by davidfowl** [comment]
I’m not sure if it’s possible to get the replica id for the running docker compose instance (that would be ideal). In kubernetes I’d assume we want the pod name (for the service instance id)

**Comment by martinjt** [comment]
I'm fairly certain that the `service.instance.id` defaults, I could be wrong there.

The issue is more that the dashboard treats the instance name as human readable, rather than a guid, making the UI hard to navigate.  The `service.name` resource attribute is the thing that defines the service, the instance is "which one of the many instances of that service did the request use".

The other thing I'd mention on the collector is that from a K8s perspective, there are well established patterns for deployment (daemonset on the node, then adding the environment variables to forward service traffic to the node IP rather than a k8s service). So I'm not sure it would be aspirate's responsibility to set those up, unless it's also creating the cluster. More that you should consider "where" the collector lives in that environment.

I would add some other defaults that might be good. Things like `deployment.type=aspire` would be a good one, maybe add other things like the commit from the apphost etc.

**Comment by prom3theu5** [comment]
Yeah I think the best course there certainly for inclusion of the collector would be to assume that it's part of the cluster already, as in externally managed separate of aspirate and prompt for it during first run of generate.

Two prompts. To you want to include and then specify endpoint. 
Unless it exists already as an env var defined in the manifest, then skip asking and use the exposed endpoint verbatim

This would be a first time prompt only as once set it'd persist in the aspirate state file

**Comment by prom3theu5** [comment]
> I’m not sure if it’s possible to get the replica id for the running docker compose instance (that would be ideal). In kubernetes I’d assume we want the pod name (for the service instance id)

we'd probably need to use metadata.uid in k8s
In docker - we could use the generated hostname (possible) - it has to be something in both cases though that we can use at runtime, without needing a custom entrypoint

**Comment by prom3theu5** [comment]
@davidfowl 
Inclusion of the service name has just gone out in v0.6.13-preview
This does fix the dashboard display in kube and compose

Replica isolation will still be an issue. I'll leave this issue open until we also implement a simple way to switch between the dashboard and the otlp collector endpoint, its the same env var that needs to be set isn't it?
We've also other routes to explore there, like side-car injection etc but that may be beyond the scope of this issue

![image](https://github.com/prom3theu5/aspirational-manifests/assets/1518610/44f3db14-ef28-46bb-9f9b-080fe1c5826c)

![image](https://github.com/prom3theu5/aspirational-manifests/assets/1518610/1b1082c3-f3ab-4298-8a5e-b7fbba1f28af)

**Comment by coen22** [comment]
When I run Aspire HTTPS directly, I'm also seeing the Resources and Console tab. Is it possible to have those as well with the container?
